### DISCUSSION TITLE
Your proposed upload is larger than the maximum object size specified in your Policy Document

### DISCUSSION THREAD

**Discussion by SandroMaglione** [vercel/next.js#23314] [Help] [answered]
Hi,

I am trying to deploy my next js application to Firebase. My configuration basically uploads the static files into Firebase Hosting. It then creates a single Firebase function for SSR.

When I try to deploy the function, its size is `100.33 MB`:
```
i  functions: packaged . (100.33 MB) for uploading
```
The size is [too large for Firebase Functions](https://firebase.google.com/docs/functions/quotas#resource_limits), and I got the following error:
```
<?xml version='1.0' encoding='UTF-8'?><Error><Code>EntityTooLarge</Code><Message>Your proposed upload is larger than the maximum object size specified in your Policy Document.</Message><Details>Content-length exceeds upper bound on range</Details></Error>

!  functions: Upload Error: HTTP Error: 400, Unknown Error
```

The following is my configuration file to generate the function:
```js
const { https } = require('firebase-functions');
const { default: next } = require('next');

const isDevelopment = process.env.NODE_ENV !== 'production';

const server = next({
  dev: isDevelopment,
  conf: { distDir: '.next' },
});

const nextjsHandle = server.getRequestHandler();
exports.nextServer = https.onRequest((request, response) => {
  return server.prepare().then(() => nextjsHandle(request, response));
});

```

How is it possible that a single function takes 100MB of space? Is it possible to optimize its size? How can I reduce its size?

**Comment by akellbl4** [comment, marked as answer]
Hi
I'm sure you should upload `node_modules` to firebase. Try to check if any dev modules are there. You don't need them in production.
My `.next` folder is 236MB but it contains pictures and videos. A single function should weigh much less.

**Reply by SandroMaglione** [reply]
I think I am doing something wrong probably.

My `.next` folder is `394MB` and my project does not have images or videos. My total bundle size is `2.61MB` using `webpack-bundle-analyzer`.

This is the output of the build command:
```
First Load JS shared by all                                  133 kB
  ├ chunks/1921d899ccd22b7297ca7390febf79b4d62b04e7.cf428f.js  13.7 kB
  ├ chunks/6c70b1170c727eb6e729f439eba5967804394444.8122ed.js  4.32 kB
  ├ chunks/7e76759a92ae911724ac8c047bc5f23c66830edb.06b00c.js  40.4 kB
  ├ chunks/c8e3823e51a23f8fda9c882e5cebdbf2c25a08b6.98a859.js  14.3 kB
  ├ chunks/cce79cc5f39e8795d1db7249b0e50a2057f3680b.a7e9b7.js  6.57 kB
  ├ chunks/framework.0c2392.js                                 42.1 kB
  ├ chunks/main.a72562.js                                      6.76 kB
  ├ chunks/pages/_app.f95088.js                                4.12 kB
  ├ chunks/webpack.44673d.js                                   1.21 kB
  └ css/65d7f423e8885e0778a9.css                               7.35 kB

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
   (ISR)     incremental static regeneration (uses revalidate in getStaticProps)
```

Any idea on what I am missing?

**Reply by akellbl4** [reply]
Do you clean your `.next` folder before the production build?
I've noticed that actually fresh build of my app is `~80MB`

**Reply by SandroMaglione** [reply]
You are right.

I deleted the `.next` folder and run `npm run build` again. Now the function is around `8MB` and everything works.

As a reference, I found [this discussion](https://github.com/vercel/next.js/discussions/6009) about cleaning the build folder.
